doc: for-expr partial evaluation. (#30)
terramate-io/terramate-docs: cli/code-generation/generate-hcl.md

}
```

The `[for ...]` and `{for ...}` expressions are evaluated as much as possible, which means the "for" is only kept in the generated code if it contains unknown variables. 

See examples below:

### Full evaluation

```hcl
generate_hcl "example.hcl" {
  lets {
    numbers = [1, 2, 3]
  }

  content {
    square_values = [for x in let.numbers : x*x]
  }
}
```

Generates:
```hcl
// TERRAMATE: GENERATED AUTOMATICALLY DO NOT EDIT

square_values = [
  1,
  4,
  9,
]
```

You can also fully evaluate a loop containing `tm_hcl_expression()` calls.
Example:

```hcl
generate_hcl "example2.hcl" {
  lets {
    modules = ["module1", "module2"]
  }

  content {
    values = [for v in let.modules : tm_hcl_expression("var.${v}")]
  }
}
```
which generates:

```hcl
// TERRAMATE: GENERATED AUTOMATICALLY DO NOT EDIT

values = [
  var.module1,
  var.module2,
]
```

### Partial evaluation

> If any part of the `for` expression contains unknown variables, then the loop construct is kept in the generated code but Terramate variables are evaluated. See example below:

Reviewer: ```suggestion
If any part of the 'for' expression includes unknown variables, the loop construct stays in the generated code, while Terramate variables are evaluated. See example below:
```

Reviewer: sorry, used single quotes by mistake. It will be `for`

Reviewer: ```suggestion
If any part of the 'for' expression contains unknown variables, the loop construct stays in the generated code, while Terramate variables are evaluated. See example below:
```
